packages/mongo/collection.js

```diff
@@ -388,8 +389,7 @@ Mongo.Collection._rewriteSelector = function (selector) {
       // override the ones set on $regex.
       if (value.$options !== undefined)
         ret[key].$options = value.$options;
-    }
-    else if (_.contains(['$or','$and','$nor'], key)) {
+    } else if (_.contains(['$or','$and','$nor'], key)) {
```

Comment:
@stubailo That idiom won't do what you want if `key === "toString"` or anything else on `Object.prototype`.